packages/instrumentation/src/ActiveTracingHelper.ts

```diff
@@ -61,21 +55,23 @@ export class ActiveTracingHelper implements TracingHelper {
         }
       })
 
-      const span = new SpanConstructor(
-        tracer,
-        ROOT_CONTEXT,
+      const parentSpanContext: SpanContext = {
+        traceId: engineSpan.trace_id,
+        spanId: engineSpan.parent_span_id,
+        traceFlags: TraceFlags.SAMPLED,
+      }
+      const parentContext = trace.setSpanContext(ROOT_CONTEXT, parentSpanContext)
+      const span = tracer.startSpan(
```

Comment:
There is still one major problem with this code: the `engineSpan.spanId` is lost as we don't control the `spanId` when instanciating spans through the tracer. This is a problem as because of this the relationship between engine spans is lost.: we do keep the `parent_span_id` propagation, but this propagation is useless if the `spanId` themselves are not propagated.

I don't see an easy fix for that.
We could accumulate the engine spans, and only create them once we have a full tree of spans. That would enable us to create them from the root and remap the `parent_span_id` to the newly created ids. But I'm not sure if that would be an acceptable solution.

Reply:
That problem is, unfortunately, the main reason why we use `Span` constructor rather than `tracer`. 
The only way I could see we can solve it right now that will work with all kind of engines: ask `opentelemetry-js` people to allow to set explicit `spanId` via `tracer` or send them a pull request that would implement that.